src/components/Campaign.tsx

```diff
@@ -91,8 +91,9 @@ export const Campaign = () => {
         console.error("Only Unlayer contents can be saved as templates");
         return;
       }
-      const filename = campaignContentQuery.data?.campaignName.concat(
-        "html",
+      const campaignName = campaignContentQuery.data?.campaignName;
+      const filename = (
+        campaignName && campaignName !== "" ? campaignName : "campaign"
```

Comment:
Que te parece para evitar tambien el caso de que `campaignName` tuviese espacios en blanco, por ejemplo "      ":
```suggestion
      const filename = (campaignName?.trim() || "campaign") as string;
```